api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerExternalConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.CertAndKeySecretSource;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Configures External listeners
+ */
+
+@JsonPropertyOrder({"brokerCertAndKey"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(
+    editableEnabled = false,
+    generateBuilderPackage = false,
+    builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@EqualsAndHashCode
+public class KafkaListenerExternalConfiguration implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private CertAndKeySecretSource brokerCertAndKey;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Reference to the `Secret` which holds the certificate and private key pair.")
+    public CertAndKeySecretSource getBrokerCertAndKey() {
```

Comment:
Did we not say this could be a whole cert chain, not a single cert? In which case `getBrokerChertChainAndKey()` would be a better name.

Reply:
It is a bit tricky I guess ... `getBrokerCertMaybeWithChainAndAlwaysWithKey` :-D. But in any case it should be mentioned at least in the `@Description`.

Reply:
So the name should be `brokerCertChainAndKey` and the `@Description` should mention what? @ppatierno Can you please add your opinion?

Reply:
You should add something like _the certificate can optionally contain the whole chain_.

Reply:
@stanlyDoge I would use the @tombentley proposal and mentioning what @scholzj is suggesting in the description.